include/fuzzuf/optimizer/store.hpp

```diff
@@ -19,78 +19,108 @@
 #ifndef FUZZUF_INCLUDE_OPTIMIZER_STORE_HPP
 #define FUZZUF_INCLUDE_OPTIMIZER_STORE_HPP
 
-#include <unordered_map>
 #include <any>
 #include <string>
+#include <unordered_map>
 
 namespace fuzzuf::optimizer {
 
 /**
  * @struct StoreKey
  * @details This class is used as a key in Store.
- * If you want to add some value in Store, you should declare a new variable of StoreKey,
- * and use it as the identifier of the value.
+ * If you want to add some value in Store, you should declare a new variable of
+ * StoreKey, and use it as the identifier of the value.
  */
-template<class Type>
+template <class Type>
 struct StoreKey {
-    std::string name;
+  std::string name;
 };
 
-
 /**
  * @class Store
  * @brief Key-value store that bridges between fuzzers and optimizers
  * @note This class is implemented with the singleton pattern
- * because we want to share Store instance between fuzzers and optimizers with no difficulty.
- * This design gets devastating if we would like to have mutiple fuzzer instances in a single process/thread,
- * but we currently assume that such situations don't happen.
+ * because we want to share Store instance between fuzzers and optimizers with
+ *no difficulty. This design gets devastating if we would like to have mutiple
+ *fuzzer instances in a single process/thread, but we currently assume that such
+ *situations don't happen.
  **/
 class Store {
-public:
-    ~Store();
+ public:
+  ~Store();
 
-    static Store& GetInstance();
+  static Store& GetInstance();
 
-    template<typename Type>
-    Type Get(const StoreKey<Type>& key);
+  template <typename Type>
+  void InitKey(const StoreKey<Type>& key);
 
-    template<typename Type>
-    Type& GetMutRef(const StoreKey<Type>& key);
+  template <typename Type>
+  void InitKey(const StoreKey<Type>& key, Type val);
 
-    template<typename Type>
-    void Set(const StoreKey<Type>& key, Type val);
+  template <typename Type>
+  Type Get(const StoreKey<Type>& key, bool assert = false);
 
-    template<typename Type>
-    bool Exists(const StoreKey<Type>& key);
+  template <typename Type>
+  Type& GetMutRef(const StoreKey<Type>& key, bool assert = false);
 
-private:
-    Store();
+  template <typename Type>
+  void Set(const StoreKey<Type>& key, Type val);
 
-    std::unordered_map<std::string, std::any> data;
+  template <typename Type>
+  bool Exists(const StoreKey<Type>& key);
+
+ private:
+  Store();
+
+  std::unordered_map<std::string, std::any> data;
 };
 
-template<typename Type>
-Type Store::Get(const StoreKey<Type>& key) {
-    // may throw std::bad_any_cast
-    return std::any_cast<Type>(data[key.name]);
+template <typename Type>
+void Store::InitKey(const StoreKey<Type>& key) {
+  data[key.name] = std::any_cast<Type>(Type());
+}
+
+template <typename Type>
+void Store::InitKey(const StoreKey<Type>& key, Type val) {
+  data[key.name] = std::any_cast<Type>(Type(val));
 }
```

Comment:
If we arrange that the value contained in Store should be initialized with `InitKey` first, then probably we can check if `InitKey` is called twice with the same key. This is intended to prevent two `StoreKey<T>` variables from wrongly having the same key string like `StoreKey<int> a {"duplicated"}, b {"duplicated"};`

Reply:
fixed in 164c51cd4bee3a09c180c9591d3cf6d93c2b53b3